### ISSUE TITLE
mkdir() Permission denied

### ISSUE THREAD

**Issue by sreejith2base** [DivineOmega/laravel-password-exposed-validation-rule#7]
I am getting the following error. 

My Laravel version 5.7, PHP 7.2  
![error](https://user-images.githubusercontent.com/14835876/48202466-f6c68f00-e38a-11e8-907b-b36de8e4e1b0.png)

**Comment by DivineOmega** [comment]
Thanks for reporting this issue.

It seems your system temporary directory does not have sufficient permissions. Make sure it is writable and try again.

Your system temporary directory (as reported by the `sys_get_temp_dir()` method) is usually just `/tmp` on Linux.

Give me a shout if you're still having issues later.

**Comment by rutgergrasgroen** [comment]
Hi,

I have the same problem. If you have a staging and a production subscription on the same server, then the 'password-exposed-cache' folder is permitted to one user (in my case staging). The production user cant write in the 'password-exposed-cache'. 

Is it possible to fix that?

**Comment by DivineOmega** [comment]
It's possible to pass a custom instance of `PasswordExposedChecker` to the `PasswordExposed` rule when you create it. This will allow you specify a custom cache pool object, and thus change the cache directory.

Something similar to the following should do the trick.

```php
use \DivineOmega\DOFileCachePSR6\CacheItemPool;
use \DivineOmega\PasswordExposed\PasswordExposedChecker;
use \DivineOmega\LaravelPasswordExposedValidationRule\PasswordExposed;

$cache = new CacheItemPool();
$cache->changeConfig([
    'cacheDirectory' => sys_get_temp_dir().'/my-custom-cache-directory/',
]);

$passwordExposedChecker = new PasswordExposedChecker(null, $cache);

$request->validate([
    'password' => ['required', new PasswordExposed($passwordExposedChecker)],
]);
```

**Comment by DivineOmega** [comment]
In fact, I may modify this library to store the cache within the Laravel project `storage` directory. This should prevent the issue of multiple site's caches conflicting in the manner you are seeing @rutgergrasgroen.

**Comment by rutgergrasgroen** [comment]
Thanks, that would be great!

**Comment by DivineOmega** [comment]
This should be fixed in v1.2.1. Please give it a test and re-open the issue if you have any problems. :slightly_smiling_face:
